### DISCUSSION TITLE
subprocess output in TextLog or TabbedContent?!?

### DISCUSSION THREAD

**Discussion by jedie** [Textualize/textual#2222] [General]
I have to call a few `subprocess.check_call()` that generates many text output lines. Is it possible to "redirect" these output into a `TextLog` or `TabbedContent` ?

The idea is to display the output, but in a way that the terminal is not flooded with text lines.

Maybe in combination with https://pypi.org/project/subprocess-tee ?

Any idea?

**Comment by 20urc3** [comment]
up

**Comment by huiyeruzhou** [comment]
you can catch a print event and use a RichLog or something else to show it, 

`self.begin_capture_print(self, True, True) `

in on_mount , and

```
    @on(events.Print)
    def log_printed(self, event: events.Print):
        if not event.text:
            return
        time_str = datetime.datetime.now().strftime('%H:%M:%S')
        self.query_one(".logger", RichLog).write(f"[{time_str}] {event.text}")
```
